src/sentry/taskworker/consumer_grpc.py

```diff
@@ -0,0 +1,57 @@
+import logging
+import time
+
+import grpc
+from sentry_protos.sentry.v1alpha.taskworker_pb2 import (
+    TASK_ACTIVATION_STATUS_PENDING,
+    DispatchRequest,
+)
+from sentry_protos.sentry.v1alpha.taskworker_pb2_grpc import WorkerServiceStub
+
+from sentry.taskworker.pending_task_store import PendingTaskStore
+
+logger = logging.getLogger("sentry.taskworker.grpc_server")
+
+
+class ConsumerGrpc:
+    def __init__(self) -> None:
+        self.pending_task_store = PendingTaskStore()
+        self.host = "localhost"
+        self.server_port = 50051
+        self.channel = grpc.insecure_channel(f"{self.host}:{self.server_port}")
+        self.stub = WorkerServiceStub(self.channel)
+
+    def start(self):
+        while True:
+            self.dispatch_task()
```

Comment:
We're still polling/pulling here, maybe we should have a multithreading pool here with async grpc. Put our ongoing connection with the worker into the threadpool, when any of them completes, we only query the db then?

Reply:
I think multithreading is going to be necessary. I think we'll want to pair the threadpool count to be <= the number of workers per partition-consumer.  Having more threads than workers feels like it would lead to contention on workers.